### DISCUSSION TITLE
Bundle size

### DISCUSSION THREAD

**Discussion by luwes** [nanojsx/nano#94] [Help / Questions]
Really awesome lib, great work! Was just wondering, is the 1kB still accurate? 

The core bundle is more like 3kB from my test.

```
cat /Users//Sites/other/dom/nano/bundles/nano.core.min.js | gzip -c9 | wc -c
2984
```

**Comment by yandeu** [comment]
Thanks! At the moment, there is a lot of server-side stuff in the core, which is also bundled on the client. I could split it, but since the code works isomorphic it is much easier to just have only file. 

What to you think? Do those few bytes matter?

Maybe we could tell Webpack to omit the server-side code? _I have never done this_.

Edit: I guess this wouldn't even be useful, since most user include the lib via npm. Other suggestions?

**Reply by luwes** [reply]
I guess it depends on the amount of bytes 😬 

It's a bit hard to give a recommendation because I haven't done SSR in my lib but I would try not to ship too much code meant for the server to the client. Maybe check another lib how they do it, Solid JS works a bit more like this one I think. No vdom.

**Reply by yandeu** [reply]
I just reduced the bytes from 3031 to only 1753 in https://github.com/nanojsx/nano/commit/1ab0348ba7c28eab12b91b4baadd5e8dc43a4eb5, by moving a lot of SSR stuff.

Also the badge "gzip size" in the README file should indicate the correct size of the core module. It will be about 1.75 in the next release.

Not sure if I can reduce it much more. In Nano JSX the core is responsible for a lot. Handling SVGs, WebComponent, Events, JSinCSS, rendering Functional and Class Components etc.

And if you look at `nano.slim.min.js`, it is 6653 bytes and has all the features from core + Class Component, Fragment, Store, State, Context, withStyles(), Router, and you can use jsx as tagged template literals instead of using h.

**Reply by jrson83** [reply]
Nice. Try using this zopfli repo for gzip:

https://github.com/MrKrzYch00/zopfli

It  can save up to 40%:

https://minime.stephan-brumme.com/bootstrap/5.1.0/

**Reply by yandeu** [reply]
The code is not shipped compressed. This is the responsibility of the server. The bundle is only compressed (using gzip) for measurement.

**Reply by jrson83** [reply]
Well I know, but you could provide the info of the file size when using zopfli additionally.
